### PR TITLE
[storybook-v2] Initial setup of new Storybook

### PR DESCRIPTION
**Pull request by viliusmileris**
What was done:
* Deleted old Storybook project
* Created Storybook Addon panel to preview RicosEditor content state. It works both ways - changes made in RicosEditor will be presented in Content Addon and vice versa. Also, content sate is being saved in query params.

Regarding Docusaurus not building docs issue:
`"export 'mdx' (imported as 'mdx') was not found in '@mdx-js/react' (possible exports: MDXContext, MDXProvider, useMDXComponents, withMDXComponents)"
`
As a workaround I've added this commit https://github.com/wix/ricos/pull/5067/commits/bf95dee96775405c3df982c29a0e37d097cb6f63 which overrides `@mdx-js/react@2.0` version with `@mdx-js/react@1.6.22` , because looks like `@storybook/addon-docs` is using this import `import { mdx } from '@mdx-js/react'` which doesn't exist in the latest MDX version.

### CODE REVIEW

File: packages/tiptap-extensions/src/ricos-styles.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import type { CommandProps, ExtensionConfig } from '@tiptap/core';
 import type { Node as ProseMirrorNode } from 'prosemirror-model';
 import { omit } from 'lodash';
 
-const { Step, StepResult } = require('prosemirror-transform');
+import { Step, StepResult } from 'prosemirror-transform';
```
Comment (viliusmileris) [inline packages/tiptap-extensions/src/ricos-styles.ts:17]: vite throws errors while using commonjs imports

File: examples/storybook/src/addons/addon-content/components/ContentPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    setState((prevState): PanelState => {
+      if (!isRichContent(c)) {
+        return prevState;
+      }
+
+      if (prevState.type === 'default' || prevState.type === 'updated') {
+        return {
+          type: 'updated',
+          defaultValue: prevState.defaultValue,
+          componentName: prevState.componentName,
```
Comment (dragonofdeath) [inline examples/storybook/src/addons/addon-content/components/ContentPanel.tsx:80]: maybe we can drop this componentName as its in contentOverrides already?

File: examples/storybook/src/addons/addon-content/hooks/usePluginGlobals.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { isEqual } from 'lodash';
+import React from 'react';
+import { encodeContentGlobal, parseContentGlobal } from '../parseGlobal';
+import { GLOBALS_KEY } from '../constants';
+import type { ContentGlobal } from '../models/Content';
+
+export const usePluginGlobals = (globals, updateGlobals) => {
+  // We dont't want `update` reference changes on globals update, so we store globals in ref
+  const tpaSettingsGlobalsRef = React.useRef<ContentGlobal | undefined>();
```
Comment (dragonofdeath) [inline examples/storybook/src/addons/addon-content/hooks/usePluginGlobals.ts:9]: what is tpaSettings?

**Review by dragonofdeath — APPROVED** [review]

File: examples/storybook/src/addons/addon-content/components/ContentPanel.tsx (hunk 2 above)
Comment (viliusmileris) [reply examples/storybook/src/addons/addon-content/components/ContentPanel.tsx]: true, removed

**Review by domakas — APPROVED** [review]

**Pull request merged**